src/MongoDB.Driver.Core/Core/Servers/RoundTripTimeMonitor.cs

```diff
@@ -0,0 +1,166 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Diagnostics;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Driver.Core.Connections;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Core.Servers
+{
+    internal interface IRoundTripTimeMonitor : IDisposable
+    {
+        TimeSpan Average { get; }
+        void AddSample(TimeSpan roundTripTime);
+        void Reset();
+        Task RunAsync();
+    }
+
+    internal class RoundTripTimeMonitor : IRoundTripTimeMonitor
+    {
+        private readonly ExponentiallyWeightedMovingAverage _averageRoundTripTimeCalculator = new ExponentiallyWeightedMovingAverage(0.2);
```

Comment:
In case anyone is wondering why an alpha of 0.2 was chosen:
https://github.com/mongodb/specifications/blob/4e570fc442e0a67bcb4999c85a0afb10c0c5a8ae/source/server-selection/server-selection.rst#calculation-of-average-round-trip-times
```
> A weighting factor of 0.2 was chosen to put about 85% of the weight of the average RTT on the 9 most recent observations.
```

Reply:
Nit: should this weighing factor be defined as a constant?